doc/changelog.txt

```diff
@@ -32,6 +32,63 @@ BREAKING CHANGES & HOW TO GET BACK OLD BEHAVIOURS:
      return originalGetConfigInt(key, def) or 0
    end
 
+ ! hovercraft and ships brought out of water no longer forced to be upright.
+   To get back previous behaviour, put the `upright = true` tag in all unit defs
+   whose move def is of the hovercraft or ship type.
+
+ ! The return value from the UnitUnitCollision callin is now ignored
+   and there is only one event for each collision.
+   There is no way to get back the old behaviour for now,
+   but if someone needs it it could be arranged.
+
+ ! The GameProgress callin is now called every 5 seconds (from 10).
+   This value is not exposed anywhere so to get back the previous
+   behaviour you will have to find whichever code assumes the old value.
```

Comment:
Technically it's easy to write code giving previous behaviour by just skipping every other call, but that's dumb.

Reply:
It's just an unconditional change that doesn't presume any workarounds to return to the prev. defaults.

Reply:
Nah, I mean for a naive calculation. Consider this:
```lua
local previousLocalFrame = 0
local FRAMES_PER_EVENT = 300 -- but really 150 in new engine
function widget:GameProgress()
    local currentLocalFrame = Spring.GetGameFrame()
    local howManyFramesDidWeManageToSimulateLocally = currentLocalFrame - previousLocalFrame
    if howManyFramesDidWeManageToSimulateLocally < FRAMES_PER_EVENT then
      Spring.Echo("We are catching up slower than the server simulates, we will never catch up!")
      ShowWarningWindow("you will never catch up, resign lobster")
      -- but maybe we simulated 290 locally and are still catching up at ~2x speed in reality
   end
   previousLocalFrame = currentLocalFrame
end
```